src/cdomains/addressDomain.ml

```diff
@@ -133,7 +133,7 @@ struct
     | false, false -> cop x y
 
   let meet x y   = merge join meet x y
-  let narrow x y = merge widen narrow x y
+  let narrow x y = merge (fun x y -> widen x (join x y)) narrow x y
```

Comment:
Should this be fixed here where it's just applied or further up?
`widen` seems to be included from `HoareDomain.HoarePO`.

Reply:
The question is whether the `widen` operations in general can expect that the first value is less or equal to the second. If yes, then performing a join on the second operand in the `widen` seems redundant.

Reply:
We join in the solver, i.e. at the very top. So either it really is required here due to some complicated merge/join/widen interaction or a join is not done somewhere else.

Reply:
> The question is whether the widen operations in general can expect that the first value is less or equal to the second.

Yes, I thought that's what we're going for.

Reply:
> We join in the solver, i.e. at the very top. So either it really is required here due to some complicated merge/join/widen interaction or a join is not done somewhere else.

https://github.com/goblint/analyzer/blob/e00c961b38cb4bd2e7a417cab576de9f9d9acf46/src/cdomains/addressDomain.ml#L124-L136

We are inside a `narrow` here where we want to have special handling if both pointers contain the top pointer (namely not throw away definite addresses). So it's not a place where the solver calls `widen` it is just us manually deciding we want to do that in some cases instead of a `narrow`. So I think we need to fix it here.